Docs/Tutorial.md

```diff
@@ -0,0 +1,146 @@
+- [Adding a Configuration File](#adding-a-configuration-file)
+- [Updating the Code](#updating-the-code)
+- [Next Steps](#next-steps)
+
+## Getting Started
+
+If you haven't already heard about CreateAPI, it's a tool that allows you to generate Swift code from an OpenAPI specification.
+
+For a new project, the easiest way to get started with CreateAPI is to generate the entire package using the `--package` option. Once you have your package in place, you can build upon it yourself and if you need to update the generated entities or paths later on, you can do so using the `--module` option to avoid overwriting the rest of your changes.
+
+If you haven't already, be sure to [install](../README.md#installation) the `create-api` CLI.
+
+## Generating a Package
+
+First things first, you need to generate the package. To do so, there is one key ingredient: The schema. Download the Petstore schema from [https://petstore3.swagger.io/api/v3/openapi.json](https://petstore3.swagger.io/api/v3/openapi.json) and save it somewhere accessible on your computer.
+
+Using terminal, run the `create-api generate` command and specify the following:
+
+1. The `--package` option, followed by the name of your package
+2. The `--output` option, followed by the directory that you want to write the package to
+3. The path to the schema that you just downloaded
+
+For example:
+
+```bash
+$ create-api generate --package "PetstoreKit" --output "./" "path/to/the/downloaded/schema.json"
+```
+
+You will see something like the following printed to the console:
+
+> Generating code for schema.json...
+
+And you will find a new folder called **PetstoreKit** located within the output directory. Double click on the **Package.swift** file within the package to open it in Xcode and you will see something like the following:
+
+<img
+  src="./Images/Tutorial_01_NewPackage.png#gh-light-mode-only"
+  alt="A screenshot of Xcode focusing on the Paths.swift source file within the newly generated PetstoreKit package. To the left, you can see the project navigator which displays the source files (Entites.swift, Paths.swift) and the resolved dependencies (Get, HTTPHeaders, URLQueryEncoder)."
+/><img
+  src="./Images/Tutorial_01_NewPackage_Dark.png#gh-dark-mode-only"
+  alt="A screenshot of Xcode focusing on the Paths.swift source file within the newly generated PetstoreKit package. To the left, you can see the project navigator which displays the source files (Entites.swift, Paths.swift) and the resolved dependencies (Get, HTTPHeaders, URLQueryEncoder)."
+/>
+
+Once the dependencies have resolved, build the project and everything will succeed. Congratulations, you've generated your first project using CreateAPI!
+
+## Using the Code
+
+Now that you have the generated Paths and Entities, you probably want to actually put them to good use!
+
+In the **Sources** directory, add a new file called **PetstoreClient.swift**. Within that file, add the following code:
```

Comment:
I kind of think you should add a note here about how you don't have to modify the generated package, you can instead include it in your own project. For APIs that are in development simultaneously to the Swift project, you'll probably be generating the package over and over again during development, so modifying it directly is probably not a great idea.

This leads me to my next thought, which is maybe there could be a "best practices" document with some tips and tricks in it? A couple of ideas for that:

1. Put your CreateAPI generation command in a bash script that's checked into git somewhere, (along side the generated package), so it's easy for anyone on the project to update the API by running the script.
2. A note about writing tests that check for all the properties in the generated package that you are using specifically, so when the API is updated you can run your tests and make sure the API still gives you back the objects and properties you are using in your application.

Reply:
I see that you do address overwriting changes to the generated package when talking about modules. I still think a note about using the package as opposed to modifying it here would be nice.

Reply:
Since package is the default/recommended option (AFIAR), ideally you should never modify its contents. One of the main initial goals was to provide enough tools to generate code that compiles and you are happy with, making this workflow (always autogenerate) feasible.

Reply:
There is also a `--watch` option (for autogeneration) which I'm not sure is documented.

Reply:
So the aim of from the tutorial was to demonstrate both approaches to help you understand what is possible. It wasn't necessarily to say "you must do it like this" but I guess that I can make that a bit clearer 👍

>  I still think a note about using the package as opposed to modifying it here would be nice.

I'll do this then 👍 

> Since package is the default/recommended option (AFIAR), ideally you should never modify its contents. One of the main initial goals was to provide enough tools to generate code that compiles and you are happy with, making this workflow (always autogenerate) feasible.

Yup that's cool.

From trying things out, I guess the current issue I found with how CreateAPI works today is that you can't add extra targets or tests. The idea of having CreateAPI generate the whole package is great, but it means that the rest of your project has to go in a separate project/package where traditionally you might have just had them as two targets in the same package.  

For example, I personally use CreateAPI today in `--module` mode to integrate it into my existing networking client which has to do a lot of other stuff (mostly related to auth). [appstoreconnect-swift-sdk](https://github.com/avdLee/appstoreconnect-swift-sdk) also does something very similar. 

I'll add a note, but I agree that it would be great to revisit this in the future once we've built on the current capabilities somewhat. 

For example, I had an shower thought that maybe we could benefit from a `create-api init` command that offers a guided process to setup a package (repo), pull the schema into the right location, set a default config and even integrate a Swift Package Plugin to enable easy regeneration each time its required 😬

Reply:
> in a separate project/package where traditionally you might have just had them as two targets in the same package

That just goes to show again that as someone who is not actively using the tool, it was a great idea to ask for help to maintain it. I was thinking about a scenario where you have an Xcode Project, you generate the package in one of the subfolders, and then integrate it as a local dependency.

Reply:
> I was thinking about a scenario where you have an Xcode Project, you generate the package in one of the subfolders, and then integrate it as a local dependency.

To be honest, it does sound like a perfect use case too so I think it's a great idea supporting both. The majority of my SPM experience comes from outside of macOS/iOS Xcode Projects so I haven't thought much about this before. 

---

I plan to add the above disclaimer to the document:

```suggestion
> **Note**: In some cases, you might prefer to use the generated Package as-is by referencing it as a dependency in another project or package (this allows you to regenerate it whenever the spec changes), but for the educational purposes in this tutorial you will work directly inside the package instead.

In the **Sources** directory, add a new file called **PetstoreClient.swift**. Within that file, add the following code:
```

Reply:
> I was thinking about a scenario where you have an Xcode Project, you generate the package in one of the subfolders, and then integrate it as a local dependency.

This is definitely how I'm using it currently. Personally, I love that it's a separate package and import.

Reply:
Good to know thank you. I think there certainly is still room for another document that lists out various different possibilities explaining how you can go about using CreateAPI. It'll even be useful for me/other maintainers 😄 

I'll probably leave it out of this PR for now, but might come back to it after if somebody else doesn't beat me to it.

Reply:
So I've actually revisited this when updating everything to reflect the changes in 0.1.0. I decided to focus the tutorial entirely on generating packages and make no mention of the alternative approach. 

Later on I will add to the **AdvancedSetup.md** with instructions on how to integrate into an existing project 👍

Reply:
I went and took a look at your new Tutorial, and it's _fantastic_! Really excellent work! 🤩